base/array.jl

```diff
@@ -387,8 +387,22 @@ julia> fill(1.0, (5,5))
  1.0  1.0  1.0  1.0  1.0
 ```
 
-If `x` is an object reference, all elements will refer to the same object. `fill(Foo(),
-dims)` will return an array filled with the result of evaluating `Foo()` once.
+If `x` is an object reference, all elements will refer to the same object.
```

Comment:
I don't really understand the original comment here. What is an `object reference`? Feels like this should just be

```
!!! note
   All elements in the resulting array will refer to the same object.
```